bitsail-connectors/connector-druid/src/main/java/com/bytedance/bitsail/connector/druid/sink/DruidWriter.java

```diff
@@ -0,0 +1,267 @@
+    mapper.configure(MapperFeature.AUTO_DETECT_IS_GETTERS, false);
+    mapper.configure(MapperFeature.AUTO_DETECT_SETTERS, false);
+    mapper.configure(SerializationFeature.INDENT_OUTPUT, false);
+    mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+    return mapper;
+  }
+
+  private DataSchema provideDruidDataSchema(final BitSailConfiguration writerConfig) {
+    final String datasource = writerConfig.getNecessaryOption(DATASOURCE, REQUIRED_VALUE);
+    final List<DimensionSchema> dimensionSchemas = this.columnInfos.stream()
+            .map(this::transformToDimensionSchema)
+            .collect(Collectors.toList());
+    return new DataSchema(
+            datasource,
+            new TimestampSpec(TIMESTAMP_SPEC_COLUMN_NAME, "auto", null),
+            new DimensionsSpec(dimensionSchemas),
+            null,
+            new UniformGranularitySpec(Granularities.HOUR, Granularities.MINUTE, false, null),
+            null
+    );
+  }
+
+  private DimensionSchema transformToDimensionSchema(final ColumnInfo columnInfo) {
+    final String columnName = columnInfo.getName();
+    final String columnType = columnInfo.getType().toLowerCase();
+    final TypeInfo<?> typeInfo = typeInfoConverter.fromTypeString(columnType);
+    final DimensionSchema dimensionSchema;
+    if (TypeInfos.STRING_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new StringDimensionSchema(columnName);
+    } else if (TypeInfos.FLOAT_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new FloatDimensionSchema(columnName);
+    } else if (TypeInfos.DOUBLE_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new DoubleDimensionSchema(columnName);
+    } else if (TypeInfos.LONG_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new LongDimensionSchema(columnName);
+    } else {
+      throw new BitSailException(UNSUPPORTED_OPERATION,
+              "Column type name " + columnName + " type " + columnType + " is not supported.");
+    }
+    return dimensionSchema;
+  }
+
+  @VisibleForTesting
+  ParallelIndexIOConfig provideDruidIOConfig(final StringBuffer data) {
+    final List<String> formatList = columnInfos.stream().map(ColumnInfo::getName).collect(Collectors.toList());
+    formatList.add(TIMESTAMP_SPEC_COLUMN_NAME);
+    return new ParallelIndexIOConfig(
+            null,
+            new InlineInputSource(data.toString()),
+            new CsvInputFormat(formatList, DEFAULT_LINE_DELIMITER, null, false, 0),
+            false,
+            null
+    );
+  }
+
+  @VisibleForTesting
+  ParallelIndexSupervisorTask provideIndexTask(final ParallelIndexIOConfig ioConfig) {
+    return new ParallelIndexSupervisorTask(
+            null,
+            null,
+            null,
+            new ParallelIndexIngestionSpec(dataSchema, ioConfig, null),
+            null
+    );
+  }
+
+  @VisibleForTesting
+  String provideInputJSONString(final ParallelIndexSupervisorTask indexTask) throws JsonProcessingException {
```

Comment:
ditto

Reply:
Will add the following javadoc in next commit.
```
Provide JSON to be sent via HTTP request. Please see payload example in https://druid.apache.org/docs/latest/ingestion/ingestion-spec.html
```